python/_tskitmodule.c

```diff
@@ -6655,6 +6655,8 @@ TreeSequence_diversity(TreeSequence *self, PyObject *args, PyObject *kwds)
     if (parse_stats_mode(mode, &options) != 0) {
         goto out;
     }
+    /* TODO this should always be span normalised? */
+    options |= TSK_STAT_SPAN_NORMALISE;
```

Comment:
This was needed to make this correct.

I guess we need to decide if *all* stats have the `span_normalized` option. I guess so, but it's at the cost of making things more complex.